File: steit-csharp/tests/Hello.cs
using System;

using Steit.Builtins;
using Steit.Codec;
using Steit.Collections;
using Steit.State;
using Steit.State.Event;

namespace Just.To.Test {
    public sealed partial class Hello : IState {
        public Path Path { get; }

        public StateList<Int32> Numbers { get; private set; }
        public Vector<Int32> Others { get; private set; }

        public Hello(Path path = null) {
            this.Path = path ?? Path.Root;
            this.Numbers = new StateList<Int32>(this.Path.GetNested(0));
            this.Others = new Vector<Int32>(this.Path.GetNested(1));
        }

        public static event EventHandler<FieldUpdateEventArgs<StateList<Int32>, Hello>> OnNumbersUpdate;
        public static event EventHandler<FieldUpdateEventArgs<Vector<Int32>, Hello>> OnOthersUpdate;

        public static void ClearNumbersUpdateHandlers() { OnNumbersUpdate = null; }
        public static void ClearOthersUpdateHandlers() { OnOthersUpdate = null; }

        public static void ClearUpdateHandlers() {
            OnNumbersUpdate = null;
            OnOthersUpdate = null;
        }

        public static Hello Deserialize(IReader reader, Path path = null) {
            var hello = new Hello(path);
            hello.Replace(reader, shouldNotify: false);
            return hello;
        }

        public WireType? GetWireType(UInt32 tag) {
            switch (tag) {
                case 0: return WireType.Sized;
                case 1: return WireType.Sized;
                default: return null;
            }
        }

        public IState GetNested(UInt32 tag) {
            switch (tag) {
                case 0: return this.Numbers;
                case 1: return this.Others;
                default: return null;
            }
        }

        public void ReplaceAt(UInt32 tag, WireType wireType, IReader reader, bool shouldNotify) {
            switch (tag) {
                case 0: this.Numbers = this.MaybeNotify(0, StateList<Int32>.Deserialize(reader, this.Path.GetNested(0)), this.Numbers, OnNumbersUpdate, shouldNotify); break;
                case 1: this.Others = this.MaybeNotify(1, Vector<Int32>.Deserialize(reader, this.Path.GetNested(1)), this.Others, OnOthersUpdate, shouldNotify); break;
                default: reader.SkipField(wireType); break;
            }
        }

        public void ReplayListPush(IReader reader) { throw new NotSupportedException(); }
        public void ReplayListPop() { throw new NotSupportedException(); }
        public void ReplayMapRemove(UInt32 key) { throw new NotSupportedException(); }

        private TValue MaybeNotify<TValue>(
            UInt32 tag,
            TValue newValue,
            TValue oldValue,
            EventHandler<FieldUpdateEventArgs<TValue, Hello>> handler,
            bool shouldNotify
        ) {
            if (shouldNotify) {
                var args = new FieldUpdateEventArgs<TValue, Hello>(tag, newValue, oldValue, this);
                handler?.Invoke(this, args);
            }

            return newValue;
        }
    }
}

File: steit-go/example/hello.go
package main

import (
	"github.com/axieinfinity/steit-go/pkg/codec"
	"github.com/axieinfinity/steit-go/pkg/collections/vector/vec_i32"
	"github.com/axieinfinity/steit-go/pkg/path"
	"github.com/axieinfinity/steit-go/pkg/reader"
	readerpkg "github.com/axieinfinity/steit-go/pkg/reader"
	statepkg "github.com/axieinfinity/steit-go/pkg/state"
)

type Hello struct {
	Path    *path.Path
	Numbers *vec_i32.VecI32
	Others  *vec_i32.VecI32
}

func NewHello(p *path.Path, tag uint32) *Hello {
	obj := &Hello{}

	if p != nil {
		obj.Path = p
	} else {
		obj.Path = path.Root
	}

	obj.Numbers = vec_i32.NewVec(obj.Path.GetNested(0), nil)
	obj.Others = vec_i32.NewVec(obj.Path.GetNested(1), nil)
	return obj
}

func (s *Hello) Deserialize(reader reader.IReader, path *path.Path) error {
	actionMessage := NewHello(path, 0)
	statepkg.Replace(actionMessage, reader, false)
	*s = *actionMessage
	return nil
}

func (s *Hello) GetPath() *path.Path {
	return s.Path
}

func (s *Hello) GetWireType(tag uint32) *codec.WireType {
	switch tag {
	case 0:
		return codec.GetWireTypePtr(codec.WireTypeSized)
	case 1:
		return codec.GetWireTypePtr(codec.WireTypeSized)
	default:
		return nil
	}
}

func (s *Hello) GetNested(tag uint32) statepkg.IState {
	switch tag {
	case 0:
		return s.Numbers
	case 1:
		return s.Others
	default:
		return nil
	}
}

func (s *Hello) ReplaceAt(tag uint32, wireType codec.WireType, reader reader.IReader, shouldNotify bool) {
	switch tag {
	case 0:
		s.Numbers.Deserialize(reader, s.Path.GetNested(0))
	case 1:
		s.Others.Deserialize(reader, s.Path.GetNested(1))
	default:
		readerpkg.SkipField(reader, wireType)
	}
}

func (s *Hello) ReplayListPush(reader reader.IReader) { panic("") }
func (s *Hello) ReplayListPop()                       { panic("") }
func (s *Hello) ReplayMapRemove(key uint32)           { panic("") }
